File: money_exchange_rate/src/main/java/com/tools/money/ExchangeRateRepo.java
package com.tools.money;


import androidx.lifecycle.LiveData;

import com.github.leonardoxh.livedatacalladapter.Resource;
import com.tools.core.network.RetrofitProvider;
import com.tools.money.network.ConvertRateResponse;
import com.tools.money.network.CurrencyAPI;
import com.tools.money.network.CurrencyListResponse;

import java.util.Map;

import javax.inject.Inject;

public class ExchangeRateRepo {

    private CurrencyAPI currencyAPI;

    @Inject
    public ExchangeRateRepo(CurrencyAPI currencyAPI) {
        this.currencyAPI = currencyAPI;
    }

    LiveData<Resource<CurrencyListResponse>> getCurrencyList() {
        return currencyAPI.getCurrencyList();
    }

    LiveData<Resource<ConvertRateResponse>> getConvertRate(Map query) {
        return RetrofitProvider.getINSTANCE().getRetrofit().create(CurrencyAPI.class).getConvertRate(query);
    }
}

File: money_exchange_rate/src/main/java/com/tools/money/ExchangeRateRepo.kt
package com.tools.money

import com.tools.core.data.shared.PreferenceStorage
import com.tools.core.network.header.ApiHeadersProvider
import com.tools.money.network.CurrencyAPI
import javax.inject.Inject

class ExchangeRateRepo @Inject constructor(private val currencyAPI: CurrencyAPI, private val preferenceStorage: PreferenceStorage,private val apiMainHeadersProvider: ApiHeadersProvider) {

    suspend fun getCurrencyList() = currencyAPI.getCurrencyList(apiMainHeadersProvider.getAuthenticatedHeaders(preferenceStorage.accessToken))

    suspend fun getConvertRate(query: Map<String, String>) = currencyAPI.getConvertRate(apiMainHeadersProvider.getAuthenticatedHeaders(preferenceStorage.accessToken),query)

    var lastFromCurrency = preferenceStorage.fromCurrency
        set(value) {
            preferenceStorage.fromCurrency = value
            field = value
        }


    var lastToCurrency = preferenceStorage.toCurrency
        set(value) {
            preferenceStorage.toCurrency = value
            field = value
        }
}